crates/rtmp/src/error.rs

```diff
@@ -0,0 +1,59 @@
+use crate::SessionError;
+use crate::chunk::ChunkReadError;
+use crate::command_messages::CommandError;
+use crate::handshake::ComplexHandshakeError;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
```

Comment:
Its generally bad practice to call the error `Error` because it will likely collide with other error types or the `std::error::Error` trait directly. rather use `RtmpError`